### DISCUSSION TITLE
Scrubbing Very Choppy and Missing Thumbnails

### DISCUSSION THREAD

**Discussion by SBNL63** [mifi/lossless-cut#2229] [Support]
I must be doing something wrong.  The scrubbing feature is all but useless as it is so slow and choppy that it is impossible to scrub in a way that I can work with the Lossless tools.  I have a fast machine, FCP and Premiere Pro both scrub fast and smooth without any lag or choppy viewing.  Also, the thumbnails are useless, as one thumbnail shows for several minutes of video.  Why bother?

I have searched and read through the FAQs and cannot find any discussion on these topics.

I am on an iMac Pro, the latest Github version 3.63.0, even though I initially paid for and downloaded the Mac App Store version today.

What am I missing??

Thanks,
Nils

**Comment by mifi** [comment]
Hi

> The scrubbing feature is all but useless as it is so slow and choppy that it is impossible to scrub in a way that I can work with the Lossless tools

are you opening a file with an unsupported format so that it uses the ffmpeg-assisted playback? (shows in the top corner of llc)

> Also, the thumbnails are useless, as one thumbnail shows for several minutes of video. Why bother?

you can zoom in to see more thumbnails. how would you like to see the thumbnails instead?

**Comment by SBNL63** [comment]
Hi,

I created a couple of screen recordings showing LLC and Adobe Rush (FCP and PP are similar).

One shows the more choppy scrubbing along with the lack of useful thumbnails while scrubbing.

The other shows the lack of thumbnails in LLC and the very useful thumbnails in Adobe Rush at different zoom levels.

https://www.dropbox.com/scl/fo/mvwaiav7km8hkrzrumn4h/AHHq5f6hsxvL--AIw0y7vAo?rlkey=ec0jg7ny351gpyp00u8f18dou&dl=0


In my case, I have 4 of these video files of a fashion photo shoot, each video two hours long or so shot with a mounted iphone running continuously (you can easily see the concept in the Adobe Rush thumbnails).  This first video has 44 segments that I want to clip out. Now, all I would like to do is clip out the good parts of each apparel item as they were being shot.  To me, it is mind-blowing that the capability of being able to simply cut out a bunch of individual clips from one long video is so hard to find.  There should be some raw video processing tools built into all the major video editing packages that can do this lossless process.  I am glad I found LLC and that you took the time, effort, and money to build it.  Anyway, these are my thoughts and experience so far with LLC.

If there was a way to take the data from the 44 segments that I made in Adobe Rush (or FCP, PP) before I found LLC, import their IN and OUT timecodes into LLC, and create the segments that way, then I would be happy.  This of course is a work-around workflow which is never good.

Thanks for you time and help.



Best,
Nils

From: Mikael Finstad ***@***.***>
Reply-To: mifi/lossless-cut ***@***.***>
Date: Thursday, October 31, 2024 at 2:08 AM
To: mifi/lossless-cut ***@***.***>
Cc: Nils Larsen ***@***.***>, Author ***@***.***>
Subject: Re: [mifi/lossless-cut] Scrubbing Very Choppy and Missing Thumbnails (Discussion #2229)


Hi

The scrubbing feature is all but useless as it is so slow and choppy that it is impossible to scrub in a way that I can work with the Lossless tools

are you opening a file with an unsupported format so that it uses the ffmpeg-assisted playback? (shows in the top corner of llc)

Also, the thumbnails are useless, as one thumbnail shows for several minutes of video. Why bother?

you can zoom in to see more thumbnails. how would you like to see the thumbnails instead?

**Comment by mifi** [comment]
thanks for sharing. I see indeed the scrubbing is slower in losslesscut. Does this happen for all files or just a particular type of file? e.g. what if you try an mp4 file like this one: https://github.com/mifi/lossless-cut-fixtures/blob/master/hardangerbrua-dji-phantom.mp4 -  it could be a limitation in the chromium video player with that particular file type. And is it laggy when you play back also, or only when scrubbing?

As for the thumbnails, yea I see you're only getting 3 thumbnails which is unfortunate, but this is by design because the application will try to render as many thumbnails as it can within 3 seconds (up to 10 thumbs), and then it will give up. This is to not overload the CPU and not have the user wait for ages for thumbnails to appear. For your particular file on your computer the CPU is only able to render 3 thumbnails within 3 seconds. I will see if I can improve this logic a bit, but it's also supporting evidence that your computer is struggling to seek/render frames from this particular file.

As for why it's working better in FCP is likely because FCP uses some hardware decoding that LosslessCut does not support (LosslessCut is based on Chromium and therefore only supports the hardware decoding supported by it).

If you open the file in Chrome, do you see any faster scrubbing?

**Comment by SBNL63** [comment]
Hi Mikael,

Thanks.  The scrubbing seems much better in the mp4 file you gave but I don’t have or use those type.  The playback of my large file is fine, it just scrubs a bit laggy.  It is better than the very first attempt, I imagine something else may have been using CPU etc. at the time.

I understand about the thumbnails.

I’m sorry, I am not familiar with Chrome or Chromium, Chrome OS etc.  I use Google Chrome browser only when forced to, so know nothing about the open source version if that is what you mean.  I tried opening it in Google Chrome on my iMacPro, but that did not work, it just downloaded the file even though it is already on my computer.

Thanks for the help, but it seems I should just stick with trying to use FCP or PP and Rush.

Best,
Nils


From: Mikael Finstad ***@***.***>
Reply-To: mifi/lossless-cut ***@***.***>
Date: Friday, November 1, 2024 at 1:24 AM
To: mifi/lossless-cut ***@***.***>
Cc: Nils Larsen ***@***.***>, Author ***@***.***>
Subject: Re: [mifi/lossless-cut] Scrubbing Very Choppy and Missing Thumbnails (Discussion #2229)


thanks for sharing. I see indeed the scrubbing is slower in losslesscut. Does this happen for all files or just a particular type of file? e.g. what if you try an mp4 file like this one: https://github.com/mifi/lossless-cut-fixtures/blob/master/hardangerbrua-dji-phantom.mp4 - it could be a limitation in the chromium video player with that particular file type. And is it laggy when you play back also, or only when scrubbing?

As for the thumbnails, yea I see you're only getting 3 thumbnails which is unfortunate, but this is by design because the application will try to render as many thumbnails as it can within 3 seconds (up to 10 thumbs), and then it will give up. This is to not overload the CPU and not have the user wait for ages for thumbnails to appear. For your particular file on your computer the CPU is only able to render 3 thumbnails within 3 seconds. I will see if I can improve this logic a bit, but it's also supporting evidence that your computer is struggling to seek/render frames from this particular file.

As for why it's working better in FCP is likely because FCP uses some hardware decoding that LosslessCut does not support (LosslessCut is based on Chromium and therefore only supports the hardware decoding supported by it).

If you open the file in Chrome, do you see any faster scrubbing?
